### PR TITLE
Update LLVM subrepos to LLVM 15

### PR DESCRIPTION
**Pull request by arichardson**

### CODE REVIEW

**Comment by arichardson** [comment]
For some reason git-subrepo did not update the parent commit, so I edited those files manually to ensure we don't scan the entire history on the next subrepo pull.

**Review by brooksdavis — APPROVED** [review]
Spotted a couple `$FreeBSD$` lines that were reintroduced after the recent removal, otherwise looks good if it's working.

File: lib/libc++experimental/Makefile [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# $FreeBSD$
```
[the remark this answers is not in the document]
Comment (arichardson) [reply lib/libc++experimental/Makefile]: I sync'd this with the equivalent version since I assumed we'd rerun the removal scripts, but happy to drop it.

**Comment by arichardson** [comment]
merged to dev manually to avoid github rewriting commit hashes.

**Pull request closed without merging**